src/parser/smt2/smt2_cmd_parser.cpp

```diff
@@ -282,15 +282,9 @@ std::unique_ptr<Cmd> Smt2CmdParser::parseNextCommand()
       {
         d_state.checkLogicAllowsFunctions();
       }
+      // Note that we previously disallowed declare-fun in sygus here.
       // we allow overloading for function declarations
-      if (d_state.sygus())
-      {
-        d_lex.parseError("declare-fun are not allowed in sygus version 2.0");
-      }
-      else
-      {
-        cmd.reset(new DeclareFunctionCommand(name, sorts, t));
-      }
+      cmd.reset(new DeclareFunctionCommand(name, sorts, t));
```

Comment:
as a general question, what is the semantics of `declare-fun` here? The same of SMT-LIB?

Reply:
I see. Maybe it'd be good to document this somewhere? I wonder if we should have a dedicated page in the cvc5 docs to talk about SyGuS: https://cvc5.github.io/docs/cvc5-1.3.0/index.html